roxie/udplib/udpaeron.cpp

```diff
@@ -0,0 +1,418 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include <map>
+#include "jexcept.hpp"
+#include "jqueue.tpp"
+#include "udplib.hpp"
+#include "udpipmap.hpp"
+#include "udpmsgpk.hpp"
+#include "udpsha.hpp"
+#include "udptrs.hpp"
+#include "roxie.hpp"
+#include <Aeron.h>
+
+extern "C" {
+#include "aeronmd.h"
+#include "concurrent/aeron_atomic.h"
+#include "aeron_driver_context.h"
+#include "util/aeron_properties_util.h"
+}
+
+// Configurable variables  // MORE - add relevant code to Roxie
+bool useEmbeddedAeronDriver = true;
+unsigned aeronConnectTimeout = 5000;
+unsigned aeronPollFragmentsLimit = 10;
+unsigned aeronIdleSleepMs = 1;
+
+static std::thread aeronDriverThread;
+static Semaphore driverStarted;
+
+volatile bool aeronDriverRunning = false;
+
+extern UDPLIB_API void stopAeronDriver()
+{
+    AERON_PUT_ORDERED(aeronDriverRunning, false);
+    if (aeronDriverThread.joinable())
+        aeronDriverThread.join();
+}
+
+void sigint_handler(int signal)
+{
+    stopAeronDriver();
+}
+
+void termination_hook(void *state)
+{
+    stopAeronDriver();
+}
+
+inline bool is_running()
+{
+    bool result;
+    AERON_GET_VOLATILE(result, aeronDriverRunning);
+    return result;
+}
+
+int startAeronDriver()
+{
+    aeron_driver_context_t *context = nullptr;
+    aeron_driver_t *driver = nullptr;
+    try
+    {
+        if (aeron_driver_context_init(&context) < 0)
+            throw makeStringExceptionV(MSGAUD_operator, -1, "AERON: error initializing context (%d) %s", aeron_errcode(), aeron_errmsg());
+
+        context->termination_hook_func = termination_hook;
+        context->dirs_delete_on_start = true;
+        context->warn_if_dirs_exist = false;
+        context->term_buffer_sparse_file = false;
+        context->mtu_length=16384;
+        context->socket_rcvbuf=2097152;
```

Comment:
Probably. I just stole them from one of the aeron examples. It may well be that actually they should not be set at all (and therefore Aeron should be left to pick what it thinks are the optimal values).

Reply:
I did

Reply:
I did